compiler/frontend/src/org/jetbrains/kotlin/types/expressions/ExpressionTypingVisitorForStatements.java

```diff
@@ -441,6 +443,36 @@ protected KotlinTypeInfo visitAssignment(KtBinaryExpression expression, Expressi
                                   ? refineTypeByPropertyInType(bindingContext, leftOperand, leftType)
                                   : refineTypeFromPropertySetterIfPossible(bindingContext, leftOperand, leftType);
 
+
+        // Resolve assign overload
+        VariableDescriptor descriptor = BindingContextUtils.extractVariableFromResolvedCall(bindingContext, leftOperand);
+        OverloadResolutionResults<FunctionDescriptor> assignmentOperationDescriptors = new NameNotFoundResolutionResult<>();
+        KotlinType assignmentOperationType = null;
+        TemporaryTraceAndCache temporaryForAssignmentOperation = null;
+        boolean isDelegated = descriptor instanceof PropertyDescriptor && ((PropertyDescriptor) descriptor).isDelegated();
+        if (descriptor != null && !descriptor.isVar() && !isDelegated) {
+            ExpressionReceiver receiver = ExpressionReceiver.Companion.create(left, leftType, context.trace.getBindingContext());
+            temporaryForAssignmentOperation = TemporaryTraceAndCache.create(
+                    context, "trace to check assignment operation like '=' for", expression);
+            assignmentOperationDescriptors =
+                    components.callResolver.resolveBinaryCall(
+                            context.replaceTraceAndCache(temporaryForAssignmentOperation).replaceScope(scope),
```

Comment:
Do you check that `assign` fun really has `operator` modifier during call resolution?

Please add tests for this, like
```kotlin
operator fun String.assign(value: Int) {} // (1)

class Foo {
    fun String.assign(value: Int) {} // (2)

    fun test() {
        val s = "hello"
        s = 1 // should be resolved to (1)
    }
}

Reply:
Yeah, this is not checked and this test fails. It seems it's not that easy to do that for old frontend. I will have to dig into the code. Do you have any hint?

Reply:
You can start from here
https://github.com/JetBrains/kotlin/blob/master/compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/components/ResolutionParts.kt#L775

Reply:
I believe this is now resolved.